File: Demo/Sources/ViewControllers/TextView/TextViewViewController.m
//
//  Globus
//

#import "TextViewViewController.h"

@interface TextViewViewController ()

@property(nonatomic, weak) IBOutlet GLBTextView* textView;

@end

@implementation TextViewViewController

#pragma mark - UIViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    
    _textView.glb_borderColor = UIColor.darkGrayColor;
    _textView.glb_borderWidth = 1.0f;
    _textView.glb_cornerRadius = 4.0f;
    
    _textView.minimumHeight = 48.0f;
    _textView.maximumHeight = 96.0f;

    GLBTextStyle* placeholderStyle = [GLBTextStyle new];
    placeholderStyle.font = [UIFont boldSystemFontOfSize:16.0f];
    placeholderStyle.color = UIColor.lightGrayColor;
    _textView.placeholderStyle = placeholderStyle;
    _textView.placeholder = @"Placeholder";
    _textView.actionValueChanged = [GLBAction actionWithTarget:self action:@selector(changedText:)];
}

- (void)viewDidAppear:(BOOL)animated {
    [super viewDidAppear:animated];
    
    [_textView becomeFirstResponder];
}

#pragma mark - Actions

- (IBAction)pressedMenu:(id)sender {
    [self.glb_slideViewController showLeftViewControllerAnimated:YES complete:nil];
}

- (IBAction)changedText:(id)sender {
    NSLog(@"ChangedText: %@", _textView.text);
}

@end

File: Demo/Sources/ViewControllers/TextView/TextViewViewController.swift
//
//  Globus
//

import Globus

class TextViewViewController: GLBViewController {
    
    // MARK - Outlet property
    
    @IBOutlet weak var textView: GLBTextView!
    
    // MARK - UIViewController
    
    override func viewDidLoad() {
        super.viewDidLoad()
        
        self.textView.glb_borderColor = UIColor.darkGray;
        self.textView.glb_borderWidth = 1;
        self.textView.glb_cornerRadius = 4;
        
        self.textView.minimumHeight = 48;
        self.textView.maximumHeight = 96;

        let placeholderStyle = GLBTextStyle.init()
        placeholderStyle.font = UIFont.boldSystemFont(ofSize: 16)
        placeholderStyle.color = UIColor.lightGray
        self.textView.placeholderStyle = placeholderStyle
        self.textView.placeholder = "Placeholder"
        self.textView.actionValueChanged = GLBAction.init(target: self, action: #selector(changedText(_:)))
    }
    
    override func viewDidAppear(_ animated: Bool) {
        super.viewDidAppear(animated)
        
        self.textView.becomeFirstResponder()
    }
    
    // MARK - Action
    
    @IBAction internal func pressedMenu(_ sender: Any) {
        self.glb_slideViewController?.showLeftViewController(animated: true, complete: nil)
    }
    
    @IBAction internal func changedText(_ sender: Any) {
        print("ChangedText: \(self.textView.text)")
    }
    
    // MARK: - GLBNibExtension
    
    public override static func nibName() -> String {
        return "TextViewViewController-Swift"
    }
    
}
